src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java

```diff
@@ -0,0 +1,269 @@
+            "Unable to create the TextEmbedding processor as field_map is null or empty."
+        );
+
+        this.modelId = modelId;
+        this.fieldMap = fieldMap;
+        this.mlCommonsClientAccessor = clientAccessor;
+    }
+
+    private boolean checkEmbeddingConfigNotValid(Map<String, Object> fieldMap) {
+        return fieldMap.entrySet()
+            .stream()
+            .anyMatch(x -> StringUtils.isBlank(x.getKey()) || Objects.isNull(x.getValue()) || StringUtils.isBlank(x.getValue().toString()));
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) {
+        validateEmbeddingFieldsType(ingestDocument, fieldMap);
+        Map<String, Object> knnMap = buildKnnMap(ingestDocument, fieldMap);
+        try {
+            List<List<Float>> vectors = mlCommonsClientAccessor.blockingInferenceSentences(this.modelId, createInferenceList(knnMap));
+            processResponse(ingestDocument, knnMap, vectors);
+        } catch (ExecutionException | InterruptedException e) {
+            log.error("Text embedding processor failed with exception: " + e.getMessage(), e);
+            throw new RuntimeException("Text embedding processor failed with exception", e);
+        }
+        log.info("Text embedding completed, returning ingestDocument!");
+        return ingestDocument;
+    }
+
+    void processResponse(IngestDocument ingestDocument, Map<String, Object> knnMap, List<List<Float>> vectors) {
+        Objects.requireNonNull(vectors, "embedding failed, inference returns null result!");
+        log.info("Text embedding result fetched, starting build vector output!");
+        Map<String, Object> vectorOutput = buildVectorOutput(knnMap, vectors, ingestDocument.getSourceAndMetadata());
+        vectorOutput.forEach(ingestDocument::appendFieldValue);
+    }
+
+    @SuppressWarnings({ "unchecked" })
+    private List<String> createInferenceList(Map<String, Object> knnMap) {
+        List<String> texts = new LinkedList<>();
+        knnMap.entrySet().stream().filter(knnMapEntry -> knnMapEntry.getValue() != null).forEach(knnMapEntry -> {
+            Object sourceValue = knnMapEntry.getValue();
+            if (sourceValue instanceof List) {
+                ((List<String>) sourceValue).stream().filter(StringUtils::isNotBlank).forEach(texts::add);
+            } else if (sourceValue instanceof Map) {
+                processMapTypeInput(sourceValue, texts);
+            } else {
+                texts.add(sourceValue.toString());
+            }
+        });
+        return texts;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void processMapTypeInput(Object sourceValue, List<String> texts) {
+        if (sourceValue instanceof Map) {
+            ((Map<String, Object>) sourceValue).forEach((k, v) -> processMapTypeInput(v, texts));
+        } else if (sourceValue instanceof List) {
```

Comment:
When would this be a list? https://github.com/opensearch-project/neural-search/pull/18/files#diff-703b4a7aad2b342a1385adb42d1b7c2b4fa3bd778a56b35f313a4a52e77df436R94 shows its already a map?

Reply:
This would be a mixed type, e.g. 
```
{
    "key": ["v1", "v2"]
}
```